llvm/unittests/ADT/APIntTest.cpp

```diff
@@ -2918,7 +2918,7 @@ TEST(APIntTest, Average) {
   APInt A100(32, 100);
   APInt A101(32, 101);
   APInt A200(32, 200, false);
-  APInt ApUMax(32, UINT_MAX, false);
+  APInt ApUMax(32, static_cast<uint64_t>(UINT_MAX), false);
```

Comment:
How about?
```suggestion
  APInt ApUMax(32, static_cast<uint64_t>(std::numeric_limits<uint32_t>()), false);
```
or
```suggestion
  APInt ApUMax(32, (~(uint64_t(0)) >> 32, false);
```

to avoid a system-specific definition of uint?

Reply:
This would be my preferred option:
```cpp
APInt ApUMax = APInt::getMaxValue(32);
```